src/quo/components/text_combinations/username/view.cljs

```diff
@@ -0,0 +1,61 @@
+(ns quo.components.text-combinations.username.view
+  (:require [quo.components.icon :as icon]
+            [quo.components.markdown.text :as text]
+            [quo.components.text-combinations.username.style :as style]
+            [quo.foundations.colors :as colors]
+            [quo.theme]
+            [react-native.core :as rn]))
+
+(defn- username-text
+  [{:keys     [theme name-type username blur?]
+    real-name :name}]
```

Comment:
sorry I haven't seen this syntax before - `real-name :name` can you explain what is happening here? :)

Reply:
sure! :) 
It's just another way of destructuring, we can do:

```clojure
{:keys [a b c]}
```

and this is the same:

```clojure
{a :a
 b :b
 c :c}
```

in the code you mentioned, I did:

```clojure
{a         :a
 real-name :name}
```

which is extracting the `:name` key and rename the var as `real-name`, The reason is if I name it `:name`, then it'll shadow the `clojure.core/name` var, which is not a good practice.

I'd do the same (and I've done it) for some other keys like `:type` that we use very often.
Actually, I've caught code that is wrongly using `cljs.core/name` or `cljs.core/type` , since it'll compile without errors, example:

Tried to do:
```clojure
(do
  (let [{:keys [name type key]} {:name "my name"
                                 :type "some component type"
                                 :key  100}]
    [name type key]))
;; => ["my name" "some component type" 100]
```
but actually did:
```clojure
(do
  (let [{:keys [name type key]} {:name "my name"
                                 :type "some component type"
                                 :key  100}])
  [name type key])
;; => [cljs.core/name cljs.core/type cljs.core/key]
```

So I prefer for those keys:
```clojure
(let [{username  :name
       user-type :type
       user-key  :key} {:name "my name"
                        :type "some component type"
                        :key  100}]
  [username user-type user-key])

;; => ["my name" "some component type" 100]

```

Hope the explanation is clear :+1:  :smile_cat:

Reply:
thanks, I was wondering how to do this! 🙌